File: solutions/zig/nth-prime/1/nth_prime.zig
const std = @import("std");
const mem = std.mem;

fn isPrime(k: usize) bool {
    if (k <= 1) return false;
    if (k == 2 or k == 3) return true;
    if (k % 2 == 0 or k % 3 == 0) return false;
    var i: usize = 5;
    const until = 1 + std.math.sqrt(k);
    while (i < until) : (i += 6)
        if (k % i == 0 or k % (i + 2) == 0)
            return false;
    return true;
}

pub fn prime(_: mem.Allocator, number: usize) !usize {
    var i: usize = 2;
    var n = number;
    while (n > 0) : (i += 1) {
        if (isPrime(i)) n -= 1;
    }
    return i - 1;
}

File: solutions/lua/nth-prime/1/nth-prime.lua
local function is_prime(k)
  if k <= 1 then return false end
  if k == 2 or k == 3 then return true end
  if k % 2 == 0 or k % 3 == 0 then return false end
  local upto = 1 + math.sqrt(k)
  for i = 5, upto, 6 do
    if k % i == 0 or k % (i + 2) == 0 then
      return false
    end
  end
  return true
end

return function(n)
  assert(n > 0)
  local i = 2
  while n > 0 do
    if is_prime(i) then n = n - 1 end
    i = i + 1
  end
  return i - 1
end
